<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>转录组分析笔记(Me) - My Lives</title>
    <meta property="og:title" content="转录组分析笔记(Me) - My Lives">
    

    
      
    

    

    
    


<link href='//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css' rel='stylesheet' type='text/css' />



    <link rel="stylesheet" href="/blog/css/style.css" />
    <link rel="stylesheet" href="/blog/css/fonts.css" />
    <link rel="stylesheet" href="/blog/css/custom.css" />

  </head>

  
  <body class="blog">
    <header class="masthead">
      <h1><a href="/blog/">My Lives</a></h1>



      <nav class="menu">
        <input id="menu-check" type="checkbox" />
        <label id="menu-label" for="menu-check" class="unselectable">
          <span class="icon close-icon">✕</span>
          <span class="icon open-icon">☰</span>
          <span class="text">Menu</span>
        </label>
        <ul>
        
        
        <li><a href="/blog/">Home</a></li>
        
        <li><a href="/blog/en/">English</a></li>
        
        <li><a href="/blog/cn/">Chinese</a></li>
        
        <li><a href="/blog/travel/">Travel</a></li>
        
        
        </ul>
      </nav>
    </header>

    <article class="main">
      <header class="title">
      
<h1>转录组分析笔记(Me)</h1>

<h3>
  2017-11-05</h3>
<hr>


      </header>





<p>这是一个学习笔记，跟随生信技能树的学习笔记重复,把几个优秀笔记的内容重复摘录在此。</p>

<p><strong>学习提纲</strong>：<a href="http://www.biotrainee.com/thread-1750-1-1.html">RNA-seq基础入门传送门</a></p>

<p><strong>文章链接</strong>：<a href="https://www.nature.com/articles/ncomms13347">https://www.nature.com/articles/ncomms13347</a></p>

<p><strong>非常棒的学习笔记1</strong>：<a href="https://taoshengxu.github.io/DocumentGit/pdf/沈梦圆2017年转录组入门合辑0-6.pdf">PANDA姐的转录组入门（0-6）合辑</a></p>

<p><strong>非常棒的学习笔记2</strong>:<a href="https://taoshengxu.github.io/DocumentGit/pdf/浙大植物学小白的转录组笔记.pdf">浙大植物学小白的转录组笔记</a> <a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247484895&amp;idx=1&amp;sn=678da702fa929789b177d214070dd39a&amp;chksm=9b484564ac3fcc72914b0ae2c1b71adb63fb359cf7e73221be1ddfd3040efa2944c91bee8e3b&amp;mpshare=1&amp;scene=1&amp;srcid=0824NWlPEoAgwVKtIWnkEDd9#rd">Link2</a></p>

<p><strong>非常棒的学习笔记3</strong>:<a href="./post/transcriptome_analysis1.html">下一篇</a></p>

<h1 id="分析软件安装">分析软件安装</h1>

<p>最方便的安装方式就是 Anaconda</p>

<pre><code>wget https://repo.continuum.io/archive/Anaconda3-4.4.0-Linux-x86_64.sh

 conda install -c bioconda samtools=1.5
 conda install -c bioconda htseq=0.7.2
 conda install -c bioconda hisat2=2.1.0
 conda install -c bioconda fastqc=0.11.5
 conda install -c jfear sratoolkit=2.8.1
</code></pre>

<h1 id="数据下载">数据下载</h1>

<p>From NCBI GEO ftp</p>

<p>The RIP-seq an RNA-seq data have been deposited in the Gene Expression Omnibus database, with accession code <strong>GSE81916</strong></p>

<h3 id="一个python脚本下载geo数据-没有测试">一个Python脚本下载GEO数据(没有测试)</h3>

<pre><code class="language-python">#!/bin/python3
import re
import urlopen
import os
def main(geo):   
# find the FTP address from [url=https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GEO]https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GEO[/url]
   response = urlopen(&quot;https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={}&quot;.format(geo))
    pattern = re.compile(&quot;&lt;a href=\&quot;(.*?)\&quot;&gt;\(ftp\)&lt;/a&gt;&quot;)
    # use wget from shell to download SRA data 
   ftp_address = re.search(pattern,response.read().decode('utf-8')).group(1)
    os.system(' wget -nd -r 1 -A *.sra ' + ftp_address)
 
if __name__ == '__main__':
    from sys import argv
    main(argv[1])
</code></pre>

<p>保存命名为SRR_downloader.py，在命令行里运行</p>

<p>python3 SRR_downloader.py GSE81916</p>

<p>简单说明：</p>

<ul>
<li>用sys.argv从命令行中读取参数</li>
<li>用urllib.request向网页发起请求，获取response</li>
<li>用正则表达式(re)提取FTP地址</li>
<li>用os.system运行shell的命令</li>
</ul>

<h1 id="数据分析">数据分析</h1>

<h2 id="质控">质控</h2>

<pre><code>for id in `seq 56 62`do fastq-dump --gzip --split-3 -O /mnt/f/Data/RNA-Seq -A SRR35899${id}done ##很慢，建议后台多线程

##查看fastq文件
zcat SRR3589956_1.fastq.gz | head -n 4
##安装集成分析工具
conda install -c bioconda multiqc

# 先获取QC结果
ls *gz | while read id; do fastqc -t 4 $id; done
# multiqc
multiqc *fastqc.zip --pdf
</code></pre>

<p><strong>Python脚本QC汇总</strong></p>

<pre><code>import re
import zipfile
# read the zip file
def zipReader(file):
    qcfile =  zipfile.ZipFile(file)
    data_txt = [file for file in qcfile.namelist() if re.match(&quot;.*?_data\.txt&quot;, file)][0]
    data = [bytes.decode(line) for line in qcfile.open(data_txt)]
    return data
 
def fastqc_summary(data):
    module_num = 0
    bases = 0
    Q20 = 0
    Q30 = 0
    for line in data:
        if re.match('Filename', line):
            filename = line.split(sep=&quot;\t&quot;)[1].strip()
        if re.match('Total Sequence', line):
            read = line.split(sep=&quot;\t&quot;)[1].strip()
        if re.match('%GC', line):
            GC = line.split(sep=&quot;\t&quot;)[1].strip()
        if re.match(&quot;[^#](.*?\t){6}&quot;,line):
            bases = bases + 1
            if float(line.split(&quot;\t&quot;)[1]) &gt; 30:
                Q20 = Q20 + 1
                Q30 = Q30 + 1
            elif float(line.split(&quot;\t&quot;)[1]) &gt; 20:
                Q20 = Q20 + 1
 
        if re.match(&quot;&gt;&gt;END&quot;, line) :
            module_num = module_num + 1
            if module_num &gt;= 2:
                break
    Q20 = Q20 / bases
    Q30 = Q30 / bases
    summary = [filename, read, GC, str(Q20), str(Q30)]
    return summary
 
if __name__ == '__main__':
    import sys
    for arg in range(1, len(sys.argv)):
        data = zipReader(sys.argv[arg])
        summary = fastqc_summary(data)
        with open('summary.txt', 'a') as f:
            f.write('\t'.join(summary) + '\n')
</code></pre>

<pre><code class="language-shell">grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'TP53'|cut -f 1,4,5 &gt;&gt;gene.bed
grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'KRAS'|cut -f 1,4,5 &gt;&gt;gene.bed
grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'EGFR'|cut -f 1,4,5 &gt;&gt;gene.bed
bedtools igv -i gene.bed &gt;Bach_sanpshot.txt
perl -alne '{print &quot;goto $F[0]:$F[1]-$F[2]\nsnapshot $F[3].png&quot;} '
</code></pre>

<h2 id="hisat2比对">Hisat2比对</h2>

<p>HISAT2是TopHat2/Bowti2的继任者，使用改进的BWT算法，实现了更快的速度和更少的资源占用，作者推荐TopHat2/Bowti2和HISAT的用户转换到HISAT2。
官网：<a href="https://ccb.jhu.edu/software/hisat2/index.shtml">https://ccb.jhu.edu/software/hisat2/index.shtml</a></p>

<p>1.建立基因组索引or index 下载</p>

<pre><code class="language-shell">#建立基因组索引
#hisat2-build –p 4 genome.fa genome

#下载索引
cd ~/reference
mkdir -p index/hisat &amp;&amp; cd index/hisat
wget -c ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/hg19.tar.gz
wget -c ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/mm10.tar.gz
tar zxvf hg19.tar.gz
tar xvzf mm10.tar.gz
</code></pre>

<p>2.比对</p>

<pre><code class="language-shell">for i in `seq 56 58`
do
    hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome \
    -1 /data/RNASeq/fastq/SRR35899${i}_1.fastq.gz \
    -2 SRR35899${i}_2.fastq.gz \
    -S /data/RNASeq/fastq/SRR35899${i}.sam &gt; SRR35899${i}.log &amp;
done


##比对结果
xts@R710:/data/RNASeq/fastq$ for i in `seq 56 58`
&gt; do
&gt;     hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome \
&gt;     -1 /data/RNASeq/fastq/SRR35899${i}_1.fastq.gz \
&gt;     -2 SRR35899${i}_2.fastq.gz \
&gt;     -S /data/RNASeq/fastq/SRR35899${i}.sam &gt; SRR35899${i}.log &amp;
&gt; done
[1] 11177
[2] 11178
[3] 11179
xts@R710:/data/RNASeq/fastq$ tipTime loading forward index: 00:00:24
Time loading forward index: 00:00:24
Time loading forward index: 00:00:24
Time loading reference: 00:00:04
Time loading reference: 00:00:04
Time loading reference: 00:00:04
Multiseed full-index search: 00:13:22
28856780 reads; of these:
  28856780 (100.00%) were paired; of these:
    1838981 (6.37%) aligned concordantly 0 times
    24732654 (85.71%) aligned concordantly exactly 1 time
    2285145 (7.92%) aligned concordantly &gt;1 times
    ----
    1838981 pairs aligned concordantly 0 times; of these:
      90927 (4.94%) aligned discordantly 1 time
    ----
    1748054 pairs aligned 0 times concordantly or discordantly; of these:
      3496108 mates make up the pairs; of these:
        2034939 (58.21%) aligned 0 times
        1221462 (34.94%) aligned exactly 1 time
        239707 (6.86%) aligned &gt;1 times
96.47% overall alignment rate
Time searching: 00:13:26
Overall time: 00:13:50

Multiseed full-index search: 00:14:42
25914821 reads; of these:
  25914821 (100.00%) were paired; of these:
    1785160 (6.89%) aligned concordantly 0 times
    21786672 (84.07%) aligned concordantly exactly 1 time
    2342989 (9.04%) aligned concordantly &gt;1 times
    ----
    1785160 pairs aligned concordantly 0 times; of these:
      53455 (2.99%) aligned discordantly 1 time
    ----
    1731705 pairs aligned 0 times concordantly or discordantly; of these:
      3463410 mates make up the pairs; of these:
        2187330 (63.16%) aligned 0 times
        1050929 (30.34%) aligned exactly 1 time
        225151 (6.50%) aligned &gt;1 times
95.78% overall alignment rate
Time searching: 00:14:46
Overall time: 00:15:10
[1]   已完成               hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome -1 /data/RNASeq/fastq/SRR35899${i}_1.fastq.gz -2 SRR35899${i}_2.fastq.gz -S /data/RNASeq/fastq/SRR35899${i}.sam &gt; SRR35899${i}.log
[3]+  已完成               hisat2 -t -p 24 -x /data/Reference/index/hisat2/hg19/genome -1 /data/RNASeq/fastq/SRR35899${i}_1.fastq.gz -2 SRR35899${i}_2.fastq.gz -S /data/RNASeq/fastq/SRR35899${i}.sam &gt; SRR35899${i}.log
xts@R710:/data/RNASeq/fastq$ Multiseed full-index search: 00:16:08
29720636 reads; of these:
  29720636 (100.00%) were paired; of these:
    1920019 (6.46%) aligned concordantly 0 times
    25503958 (85.81%) aligned concordantly exactly 1 time
    2296659 (7.73%) aligned concordantly &gt;1 times
    ----
    1920019 pairs aligned concordantly 0 times; of these:
      61683 (3.21%) aligned discordantly 1 time
    ----
    1858336 pairs aligned 0 times concordantly or discordantly; of these:
      3716672 mates make up the pairs; of these:
        2292272 (61.68%) aligned 0 times
        1196099 (32.18%) aligned exactly 1 time
        228301 (6.14%) aligned &gt;1 times
96.14% overall alignment rate
Time searching: 00:16:12
Overall time: 00:16:36

</code></pre>

<h2 id="samtools">SAMtools</h2>

<p><a href="https://taoshengxu.github.io/DocumentGit/pdf/samtools常用命令详解.pdf">SAMtools使用手册</a></p>

<p>SAM（sequence Alignment/mapping)数据格式是目前高通量测序中存放比对数据的标准格式，当然他可以用于存放未比对的数据。</p>

<p>主要功能有：</p>

<ul>
<li><strong>samtools view : BAM-SAM/SAM-BAM 转换和提取部分比对</strong></li>
<li><strong>samtools sort : 比对排序</strong></li>
<li><strong>samtools index: 索引排序比对</strong></li>
<li>samtools merge: 聚合多个排序比对</li>
<li>samtools faidx: 建立FASTA索引，提取部分序列</li>
<li>samtools tview: 文本格式查看序列</li>
<li>samtools pileup: 产生基于位置的结果和 consensus/indel calling</li>
</ul>

<p><strong>数据转换sam-bam-sorted bam</strong></p>

<pre><code>for i in `seq 56 58`
do
    samtools view -S SRR35899${i}.sam -b &gt; SRR35899${i}.bam
    samtools sort SRR35899${i}.bam -o SRR35899${i}_sorted.bam
    samtools index SRR35899${i}_sorted.bam
done
</code></pre>

<p>SAMtools其他操作</p>

<pre><code class="language-shell">head -1000 SRR3589957.sam &gt; test.sam
samtools view -b  test.sam &gt; test.bam
samtools view test.bam | head

samtools sort test.bam -o default.bam
samtools view default.bam | head
 
# Sort alignments by leftmost coordinates, or by read name when -n is used
samtools sort test.bam default
samtools view default.bam | head


#提取1号染色体1234-123456区域的比对read
samtools view SRR3589957_sorted.bam chr1:1234-123456 | head

#在比如搭配flag(0.1.19版本没有）和flagstat，使用-f或-F参数提取不同匹配情况的read。

# 可以先用flagstat看下总体情况
samtools flagstat SRR3589957_sorted.bam

#筛选恰好配对的read,就需要用0x10
samtools view -b -f 0x10 SRR3589957_sorted.bam chr1:1234-123456  &gt; flag.bam
samtools flagstat flag.bam
</code></pre>

<h2 id="比对质控">比对质控</h2>

<ul>
<li>RSeQC——<a href="http://rseqc.sourceforge.net/">http://rseqc.sourceforge.net/</a></li>
<li>Qualimap——<a href="http://qualimap.bioinfo.cipf.es/">http://qualimap.bioinfo.cipf.es/</a></li>
<li>Picard——<a href="http://broadinstitute.github.io/picard/">http://broadinstitute.github.io/picard/</a></li>
</ul>

<p>使用RSeQC来对我们的比对结果进行质控,RSeQC包括了十多个Python脚本，实现很多功能，具体每个脚本的参数用法，都可以在官网学习.</p>

<pre><code># RSeQC的安装，需要先安装gcc；numpy；R；Python2.7
$ pip install RSeQC
# 对bam文件进行质控，其余都同样的进行
$ bam_stat.py  -i SRR3589956_sorted.bam

基因组覆盖率的QC需要提供bed文件，可以直接RSeQC的网站下载，或者可以用gtf转换
read_distribution.py -i RNA-Seq/aligned/SRR3589956_sorted.bam -r reference/hg19_RefSeq.bed
</code></pre>

<h1 id="reads-计数-from-hoptop">Reads 计数 (From hoptop)</h1>

<p>定量分为三个水平
- 基因水平(gene-level)
- 转录本水平(transcript-level)
- 外显子使用水平(exon-usage-level)。</p>

<p>1.在<strong>基因水平</strong>上，常用的软件为HTSeq-count，featureCounts，BEDTools, Qualimap, Rsubread, GenomicRanges等。以常用的HTSeq-count为例，这些工具要解决的问题就是<strong>根据read和基因位置的overlap判断这个read到底是谁家的孩子</strong>。值得注意的是不同工具对multimapping reads处理方式也是不同的，例如HTSeq-count就直接当它们不存在。而Qualimpa则是一人一份，平均分配。</p>

<p>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。如果你要比较同一个样本(within-sample)不同基因之间的表达情况，你就需要考虑到<strong>转录本长度</strong>，因为转录本越长，那么检测的片段也会更多，直接比较等于让小孩和大人进行赛跑。如果你是比较不同样本（across sample）同一个基因的表达情况，虽然不必在意转录本长度，但是你要考虑到<strong>测序深度</strong>（sequence depth)，毕竟测序深度越高，检测到的概率越大。除了这两个因素外，你还需要考虑GC%所导致的偏差，以及测序仪器的系统偏差。目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等，不同算法之间的差异与换算方法已经有文章进行整理和吐槽了。但是，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要注意你上一步所用软件会不会进行标准化。</p>

<p>2.在<strong>转录本水平</strong>上，一般常用工具为Cufflinks和它的继任者StringTie， eXpress。这些软件要处理的难题就时转录本亚型（isoforms）之间通常是有重叠的，当二代测序读长低于转录本长度时，如何进行区分？这些工具大多采用的都是expectation maximization（EM）。好在我们有三代测序。
上述软件都是alignment-based，目前许多alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献[1]指出这类方法在估计丰度时存在样本特异性和读长偏差。</p>

<p>3.在<strong>外显子使用水平</strong>上，其实和基因水平的统计类似。但是值得注意的是为了更好的计数，我们需要提供无重叠的外显子区域的gtf文件。用于分析差异外显子使用的DEXSeq提供了一个Python脚本（dexseq<em>prepare</em>annotation.py）执行这个任务。</p>

<pre><code>mkdir -p /data/RNASeq/fastq/matrix/
for i in $(seq 56 58); 
do htseq-count -f bam -r pos -s no \
/data/RNASeq/fastq/SRR35899${i}_sorted.bam /data/Reference/gtf/gencode/gencode.v26lift37.annotation.sorted.gtf \
1&gt;matrix/SRR35899${i}.count 2&gt;matrix/SRR35899${i}_htseq.log &amp;; done
</code></pre>

<h2 id="表达矩阵">表达矩阵</h2>

<p>在RNA-Seq分析中，每一个基因就是一个feature（特征？），而基因被认为是它的所有外显子的和集。在可变剪切分析中，可以单独把每个外显子当作一个feature。而在ChIP-Seq分析中，feature则是预先定义的结合域。但是确定一个read到底属于哪一个feature有时会非常棘手。</p>

<h2 id="这段描述很有意思-信息量也很多-from-hoptop">这段描述很有意思，信息量也很多(From hoptop)</h2>

<p>我们这次分析是人类mRNA-Seq测序的结果，但是我们其实只下载了3个sra文件。一般而言RNA-Seq数据分析都至少要有2个重复，所以必须要有4个sra文件才行。我在仔细读完文章的方法这一段以后，发现他们有一批数据用的是其他课题组的： For 293 cells, the mRNA-seq results of the control samples include (1) those from the doxycycline-treated parental Flp-In T-REx 293 cells by us and (2) those from the doxycycline-treated control Flp-In T-REx 293 cells performed by another group unrelated to us (sample GSM1095127 in GSE44976)。 然后和Jimmy交流之后，他也承认自己只分析了小鼠的数据，而没有分析人类的数据。所以我们需要根据文章提供的线索下载另外一份数据，才能进行下一步的分析。
这个时候就有一个经常被问到的问题：不同来源的RNA-Seq数据能够直接比较吗？甚至说如果不同来源的RNA-seq数据的构建文库都不一样该如何比较?不同来源的RNA-Seq结果之间比较需要考虑 批次效应（batch effect) 的影响。
处理批次效应，根据我搜索的结果，是不能使用FPKM/RPKM，关于这个标准化的吐槽，我在biostars上找到了如下观点：
FPKM/RPKM 不是标准化的方法，它会引入文库特异的协变量
FPKM/RPKM has never been peer-reviewed, it has been introduced as an ad-hoc measure in a supplementary 没有同行评审
One of the authors of this paper states, that it should not be used because of faulty arithmetic 作者说算法有问题
All reviews so far have shown it to be an inferior scale for DE analysis of genes Length normalization is mostly dispensable imo in DE analysis because gene length is constant
有人建议使用一个Bioconductor包<a href="http://www.bioconductor.org/packages/devel/bioc/html/sva.html">http://www.bioconductor.org/packages/devel/bioc/html/sva.html</a> 我没有具体了解，有生之年去了解补充。
还有人引用了一篇文献 IVT-seq reveals extreme bias in RNA-sequencing 证明不同文库的RNA-Seq结果会存在很大差异。
结论： 可以问下原作者他们是如何处理数据的，居然有一个居然没有重复的分析也能过审。改用小鼠数据进行分析。或者使用无重复的分析方法，或者模拟一份数据出来，先把流程走完。</p>


  <footer>
  
<nav class="post-nav">
  <span class="nav-prev">&larr; <a href="/blog/cn/2017/11/ncbi_downlaod/">NCBI-GEO数据下载</a></span>
  <span class="nav-next"><a href="/blog/cn/2017/11/transcriptome_analysis1/">转录组分析笔记（2）</a> &rarr;</span>
</nav>





<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

<script async src="//yihui.name/js/center-img.js"></script>

  



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/tex.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



  
  <hr>
  <div class="copyright">&copy; <a href="xxxx">Who care</a> 2017 - Forever, maybe</div>
  
  </footer>
  </article>
  
  </body>
</html>

