<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>转录组分析笔记 - My Lives</title>
    <meta property="og:title" content="转录组分析笔记 - My Lives">
    

    
      
    

    

    
    


<link href='//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css' rel='stylesheet' type='text/css' />



    <link rel="stylesheet" href="/blog/css/style.css" />
    <link rel="stylesheet" href="/blog/css/fonts.css" />
    <link rel="stylesheet" href="/blog/css/custom.css" />

  </head>

  
  <body class="blog">
    <header class="masthead">
      <h1><a href="/blog/">My Lives</a></h1>



      <nav class="menu">
        <input id="menu-check" type="checkbox" />
        <label id="menu-label" for="menu-check" class="unselectable">
          <span class="icon close-icon">✕</span>
          <span class="icon open-icon">☰</span>
          <span class="text">Menu</span>
        </label>
        <ul>
        
        
        <li><a href="/blog/">Home</a></li>
        
        <li><a href="/blog/en/">English</a></li>
        
        <li><a href="/blog/cn/">Chinese</a></li>
        
        <li><a href="/blog/travel/">Travel</a></li>
        
        
        </ul>
      </nav>
    </header>

    <article class="main">
      <header class="title">
      
<h1>转录组分析笔记</h1>

<h3>
  2017-11-05</h3>
<hr>


      </header>





<p>这是一个学习笔记，跟随生信技能树的学习笔记重复。</p>

<p><strong>学习提纲</strong>：<a href="http://www.biotrainee.com/thread-1750-1-1.html">RNA-seq基础入门传送门</a></p>

<p><strong>文章链接</strong>：<a href="https://www.nature.com/articles/ncomms13347">https://www.nature.com/articles/ncomms13347</a></p>

<p><strong>非常棒的学习笔记</strong>：<a href="https://taoshengxu.github.io/DocumentGit/pdf/沈梦圆2017年转录组入门合辑0-6.pdf">PANDA姐的转录组入门（0-6）合辑</a></p>

<h1 id="分析软件安装">分析软件安装</h1>

<p>最方便的安装方式就是 Anaconda</p>

<pre><code>wget https://repo.continuum.io/archive/Anaconda3-4.4.0-Linux-x86_64.sh

 conda install -c bioconda samtools=1.5
 conda install -c bioconda htseq=0.7.2
 conda install -c bioconda hisat2=2.1.0
 conda install -c bioconda fastqc=0.11.5
 conda install -c jfear sratoolkit=2.8.1
</code></pre>

<h1 id="数据下载">数据下载</h1>

<p>From NCBI GEO ftp</p>

<p>The RIP-seq an RNA-seq data have been deposited in the Gene Expression Omnibus database, with accession code <strong>GSE81916</strong></p>

<h3 id="一个python脚本下载geo数据-没有测试">一个Python脚本下载GEO数据(没有测试)</h3>

<pre><code class="language-python">#!/bin/python3
import re
import urlopen
import os
def main(geo):   
# find the FTP address from [url=https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GEO]https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GEO[/url]
   response = urlopen(&quot;https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc={}&quot;.format(geo))
    pattern = re.compile(&quot;&lt;a href=\&quot;(.*?)\&quot;&gt;\(ftp\)&lt;/a&gt;&quot;)
    # use wget from shell to download SRA data 
   ftp_address = re.search(pattern,response.read().decode('utf-8')).group(1)
    os.system(' wget -nd -r 1 -A *.sra ' + ftp_address)
 
if __name__ == '__main__':
    from sys import argv
    main(argv[1])
</code></pre>

<p>保存命名为SRR_downloader.py，在命令行里运行</p>

<p>python3 SRR_downloader.py GSE81916</p>

<p>简单说明：</p>

<ul>
<li>用sys.argv从命令行中读取参数</li>
<li>用urllib.request向网页发起请求，获取response</li>
<li>用正则表达式(re)提取FTP地址</li>
<li>用os.system运行shell的命令</li>
</ul>

<h1 id="数据分析">数据分析</h1>

<h3 id="质控">质控</h3>

<pre><code>for id in `seq 56 62`do fastq-dump --gzip --split-3 -O /mnt/f/Data/RNA-Seq -A SRR35899${id}done ##很慢，建议后台多线程

##查看fastq文件
zcat SRR3589956_1.fastq.gz | head -n 4
##安装集成分析工具
conda install -c bioconda multiqc

# 先获取QC结果
ls *gz | while read id; do fastqc -t 4 $id; done
# multiqc
multiqc *fastqc.zip --pdf
</code></pre>

<h3 id="python脚本qc汇总">Python脚本QC汇总</h3>

<pre><code>import re
import zipfile
# read the zip file
def zipReader(file):
    qcfile =  zipfile.ZipFile(file)
    data_txt = [file for file in qcfile.namelist() if re.match(&quot;.*?_data\.txt&quot;, file)][0]
    data = [bytes.decode(line) for line in qcfile.open(data_txt)]
    return data
 
def fastqc_summary(data):
    module_num = 0
    bases = 0
    Q20 = 0
    Q30 = 0
    for line in data:
        if re.match('Filename', line):
            filename = line.split(sep=&quot;\t&quot;)[1].strip()
        if re.match('Total Sequence', line):
            read = line.split(sep=&quot;\t&quot;)[1].strip()
        if re.match('%GC', line):
            GC = line.split(sep=&quot;\t&quot;)[1].strip()
        if re.match(&quot;[^#](.*?\t){6}&quot;,line):
            bases = bases + 1
            if float(line.split(&quot;\t&quot;)[1]) &gt; 30:
                Q20 = Q20 + 1
                Q30 = Q30 + 1
            elif float(line.split(&quot;\t&quot;)[1]) &gt; 20:
                Q20 = Q20 + 1
 
        if re.match(&quot;&gt;&gt;END&quot;, line) :
            module_num = module_num + 1
            if module_num &gt;= 2:
                break
    Q20 = Q20 / bases
    Q30 = Q30 / bases
    summary = [filename, read, GC, str(Q20), str(Q30)]
    return summary
 
if __name__ == '__main__':
    import sys
    for arg in range(1, len(sys.argv)):
        data = zipReader(sys.argv[arg])
        summary = fastqc_summary(data)
        with open('summary.txt', 'a') as f:
            f.write('\t'.join(summary) + '\n')
</code></pre>

<pre><code class="language-shell">grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'TP53'|cut -f 1,4,5 &gt;&gt;gene.bed
grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'KRAS'|cut -f 1,4,5 &gt;&gt;gene.bed
grep -w 'gene' gencode.v26lift37.annotation.gtf |grep -w 'EGFR'|cut -f 1,4,5 &gt;&gt;gene.bed
bedtools igv -i gene.bed &gt;Bach_sanpshot.txt
perl -alne '{print &quot;goto $F[0]:$F[1]-$F[2]\nsnapshot $F[3].png&quot;} '
</code></pre>


  <footer>
  
<nav class="post-nav">
  <span class="nav-prev">&larr; <a href="/blog/cn/2017/11/ncbi_downlaod/">NCBI-GEO数据下载</a></span>
  <span class="nav-next"><a href="/blog/cn/2017/11/transcriptome_analysis1/">转录组差异表达分析小实战</a> &rarr;</span>
</nav>





<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

<script async src="//yihui.name/js/center-img.js"></script>

  



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/tex.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



  
  <hr>
  <div class="copyright">&copy; <a href="xxxx">Who care</a> 2017 - Forever, maybe</div>
  
  </footer>
  </article>
  
  </body>
</html>

