<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>ENCODE计划数据,千人基因组计划数据,Roadmap计划数据 - My Lives</title>
    <meta property="og:title" content="ENCODE计划数据,千人基因组计划数据,Roadmap计划数据 - My Lives">
    

    
      
    

    

    
    


<link href='//cdn.bootcss.com/highlight.js/9.12.0/styles/github.min.css' rel='stylesheet' type='text/css' />



    <link rel="stylesheet" href="/blog/css/style.css" />
    <link rel="stylesheet" href="/blog/css/fonts.css" />
    <link rel="stylesheet" href="/blog/css/custom.css" />

  </head>

  
  <body class="blog">
    <header class="masthead">
      <h1><a href="/blog/">My Lives</a></h1>



      <nav class="menu">
        <input id="menu-check" type="checkbox" />
        <label id="menu-label" for="menu-check" class="unselectable">
          <span class="icon close-icon">✕</span>
          <span class="icon open-icon">☰</span>
          <span class="text">Menu</span>
        </label>
        <ul>
        
        
        <li><a href="/blog/">Home</a></li>
        
        <li><a href="/blog/en/">English</a></li>
        
        <li><a href="/blog/cn/">Chinese</a></li>
        
        <li><a href="/blog/travel/">Travel</a></li>
        
        
        </ul>
      </nav>
    </header>

    <article class="main">
      <header class="title">
      
<h1>ENCODE计划数据,千人基因组计划数据,Roadmap计划数据</h1>

<h3>
  2017-11-06</h3>
<hr>


      </header>





<p><a href="http://www.bio-info-trainee.com/1825.html">From Jimmy 原文</a>(<a href="http://www.bio-info-trainee.com/1841.html)(http://www.bio-info-trainee.com/1339.html">http://www.bio-info-trainee.com/1841.html)(http://www.bio-info-trainee.com/1339.html</a>)</p>

<p>DNA元件百科全书(Encyclopedia of DNA Elements, ENCODE)ENCODE计划的重要性我就不多说了，如果大家还不是很了解，可以直接跳到本文末尾去下载一下ENCODE教程，好好学习。该计划采用以下几种高通量测序技术来刻画了超过100种不同的细胞系或者组织内的全基因组范围内的基因调控元件信息。本来只是针对人类的，后来对mouse以及fly等模式生物也开始测这些数据并进行分析了， 叫做 modENCODE.</p>

<ul>
<li>chromatin structure (5C)</li>
<li>open chromatin (DNase-seq and FAIRE-seq)</li>
<li>histone modifications and DNA-binding of over 100 transcription factors (ChIP-seq)</li>
<li>RNA transcription (RNAseq and CAGE)</li>
</ul>

<p>目前所有数据均全部公开(<a href="http://genome.ucsc.edu/ENCODE/">http://genome.ucsc.edu/ENCODE/</a> )，ENCODE results from 2007 and later are available from the ENCODE Project Portal, encodeproject.org. 并以30篇论文在Nature、Science、Cell、JBC、Genome Biol、Genome Research同时发表(<a href="http://www.nature.com/encode">http://www.nature.com/encode</a> )。
<strong>所有数据从raw data形式的原始测序数据到比对后的信号文件以及分析好的有意的peaks文件都可以下载</strong>。</p>

<p>我这里根据自己的学习情况，简单介绍一些ENCODE计划数据下载方式，包括ENCODE官网下载,UCSC下载，ENSEMBL下载，broad研究所数据，IHEC存放的数据，还有GEO下载这6种形式！！！</p>

<h1 id="ucsc">UCSC</h1>

<p>直接浏览文件，根据文件夹分类及文件名就可以任意方式下载自己感兴趣的数据</p>

<p><a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/</a></p>

<p>大家可能会比较习惯用UCSC提供的Genome Browser工具来可视化CHIP-seq的结果，而且Genome Browser里面非常多的选项可以控制各种在线资料是否跟你的数据一起显示来做对比，所以它必然有ftp服务器存放这些数据，其中比较出名的就是ENCODE计划的相关数据啦.</p>

<h1 id="encode计划的官网下载">ENCODE计划的官网下载</h1>

<p><a href="https://www.encodeproject.org/pipelines/">https://www.encodeproject.org/pipelines/</a></p>

<p>官网的数据下载，做得像是一个购物网站，大家可以根据自己的需求把数据添加到购物篮，然后统一下载。</p>

<p>This document describes what data are available at the ENCODE Portal, ways to get started searching and downloading data, and an overview to how the metadata describing the assays and reagents are organized. ENCODE data can be visualized and accessed from other resources, including the UCSC Genome Browser and ENSEMBL.
进入 <a href="https://www.encodeproject.org/matrix/?type=Experiment">https://www.encodeproject.org/matrix/?type=Experiment</a> 可以看到里面列出了173种细胞系，148种组织，还有一堆癌症样本的，包括CHIP-seq，DNase-seq等在内的十几种高通量测序数据。</p>

<h1 id="geo">GEO</h1>

<p><a href="http://www.ncbi.nlm.nih.gov/geo/info/ENCODE.html">http://www.ncbi.nlm.nih.gov/geo/info/ENCODE.html</a></p>

<h1 id="broad-研究所托管的encode计划数据">Broad 研究所托管的ENCODE计划数据</h1>

<p><a href="http://www.broadinstitute.org/~anshul/projects/encode">http://www.broadinstitute.org/~anshul/projects/encode</a></p>

<p>原始数据在：<a href="http://www.broadinstitute.org/~anshul/projects/encode/rawdata/">http://www.broadinstitute.org/~anshul/projects/encode/rawdata/</a></p>

<h1 id="ihec">iHEC</h1>

<p><a href="http://epigenomesportal.ca/ihec/download.html">http://epigenomesportal.ca/ihec/download.html</a></p>

<p>以文件夹文件的形式直接浏览，根据自己的需求下载即可,除了ENCODE计划的数据，还有Blueprint计划和roadmap计划的数据都可以下载</p>

<h1 id="ensembl">ENSEMBL</h1>

<p><a href="http://asia.ensembl.org/info/website/tutorials/encode.html">http://asia.ensembl.org/info/website/tutorials/encode.html</a></p>

<h1 id="结尾">结尾</h1>

<p>如果你对ENCODE计划不是很了解，可以先看看一些教程：
NIH提供的ENCODE计划相关教程： <a href="https://www.genome.gov/27553900/encode-tutorials/">https://www.genome.gov/27553900/encode-tutorials/</a></p>

<p><a href="https://www.genome.gov/27562350/encode-workshop-april-2015-keystone-symposia/">https://www.genome.gov/27562350/encode-workshop-april-2015-keystone-symposia/</a></p>

<p><a href="https://www.genome.gov/27561253/encode-workshop-tutorial-october-2014-ashg/">https://www.genome.gov/27561253/encode-workshop-tutorial-october-2014-ashg/</a></p>

<p><a href="https://www.genome.gov/27553901/encode-tutorial-may-2013-biology-of-genomes-cshl/">https://www.genome.gov/27553901/encode-tutorial-may-2013-biology-of-genomes-cshl/</a></p>

<p><a href="https://www.genome.gov/27563006/encoderoadmap-epigenomics-tutorial-october-2015-ashg/">https://www.genome.gov/27563006/encoderoadmap-epigenomics-tutorial-october-2015-ashg/</a></p>

<p><a href="https://www.genome.gov/27555330/encoderoadmap-epigenomics-tutorial-october-2013-ashg/">https://www.genome.gov/27555330/encoderoadmap-epigenomics-tutorial-october-2013-ashg/</a></p>

<p><a href="https://www.genome.gov/27551933/encoderoadmap-epigenomics-tutorial-nov-2012-ashg/">https://www.genome.gov/27551933/encoderoadmap-epigenomics-tutorial-nov-2012-ashg/</a></p>

<p><a href="http://useast.ensembl.org/info/website/tutorials/encode.html">http://useast.ensembl.org/info/website/tutorials/encode.html</a></p>

<p><a href="https://www.encodeproject.org/tutorials/">https://www.encodeproject.org/tutorials/</a></p>

<p><a href="https://www.encodeproject.org/tutorials/encode-meeting-2016/">https://www.encodeproject.org/tutorials/encode-meeting-2016/</a></p>

<p><a href="https://www.encodeproject.org/tutorials/encode-users-meeting-2015/">https://www.encodeproject.org/tutorials/encode-users-meeting-2015/</a></p>

<p>DNA元件百科全书(Encyclopedia of DNA Elements, ENCODE)项目旨在描述人类基因组中所编码的全部功能性序列元件。ENCODE计划于2003年9月正式启动，吸引了来自美国、英国、西班牙、日本和新加坡五国32个研究机构的440多名研究人员的参与，经过了9年的努力，研究了147个组织类型，进行了1478次实验，获得并分析了超过15万亿字节的原始数据，确定了400万个基因开关，明确了哪些DNA片段能打开或关闭特定的基因，以及不同类型细胞之间的“开关”存在的差异。证明所谓“垃圾DNA”都是十分有用的基因成分，担任着基因调控重任。证明人体内没有一个DNA片段是无用的。</p>

<hr />

<h1 id="千人基因组计划">千人基因组计划</h1>

<p>由于时间跨度比较长，最终的数据不只是一千人，最新版共有NA编号开头的1182个人，HG开头的1768个人！它的官方网站是：有一个ppt讲得很清楚如何通过官网做的data portal来下载数据：<a href="https://www.genome.gov/pages/research/der/ichg-1000genomestutorial/how_to_access_the_data.pdf">https://www.genome.gov/pages/research/der/ichg-1000genomestutorial/how_to_access_the_data.pdf</a> 我不喜欢可视化的界面，我比较喜欢直接进入ftp自己翻需要的数据，千人基因组计划不仅仅有自己的ftp站点，而且在NCBI，EBI和sanger研究所里面也有数据源可以下载， 是非常丰富的生信入门资源！</p>

<p><a href="ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/">ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/</a>
<a href="ftp://ftp.sanger.ac.uk/pub/1000genomes/">ftp://ftp.sanger.ac.uk/pub/1000genomes/</a>
<a href="ftp://ftp.ebi.ac.uk/pub/databases/1000genomes/">ftp://ftp.ebi.ac.uk/pub/databases/1000genomes/</a>
 <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp</a></p>

<p>千人基因组计划测了5个大的人种，25个亚人种，具体介绍如下：
09/08/2014 12:00AM          1,663 20131219.populations.tsv
09/09/2014 12:00AM             97 20131219.superpopulations.tsv
其实对大部分人来说，除非你想下载千人基因组计划的原始数据来学习生物信息学分析流程，不然用不着这个ftp站点的，它自己在EBI里面的有一个非常好用的可视化界面来浏览千人基因组计划的variation结果</p>

<p>千人基因组计划 &ndash; 基因组浏览器： <a href="http://www.ncbi.nlm.nih.gov/variation/tools/1000genomes/">http://www.ncbi.nlm.nih.gov/variation/tools/1000genomes/</a>
<a href="http://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=rs35761398">http://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=rs35761398</a>  chr1:24201919:24201920
<a href="http://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?rs=2501432">http://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?rs=2501432</a>  chr1:24201920
<a href="http://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?rs=2502992">http://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?rs=2502992</a>  chr1:24201919
在千人基因组计划里面看一个rs就能看到各种人群信息：
<a href="http://browser.1000genomes.org/Homo_sapiens/Variation/Population?r=1:24201420-24202420;v=rs2501432;vdb=variation;vf=1849472">http://browser.1000genomes.org/Homo_sapiens/Variation/Population?r=1:24201420-24202420;v=rs2501432;vdb=variation;vf=1849472</a>
这些人群信息，可以画一个网路图！ 只需要变化rs ID号即可，当然并不是所有的rs ID号都在千人基因组计划里面有显示的。
还有一个java软件-可视化检测千人基因组数据</p>

<p><a href="http://bioinformatics.oxfordjournals.org/content/early/2016/03/17/bioinformatics.btw147.short?rss=1">http://bioinformatics.oxfordjournals.org/content/early/2016/03/17/bioinformatics.btw147.short?rss=1</a>
<a href="http://limousophie35.github.io/Ferret/">http://limousophie35.github.io/Ferret/</a>
但是好像不是很好用！</p>

<p>在千人基因组计划的ftp主站点里面可以下载所有数据。
<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/</a>
<a href="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/">ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/</a>
直接看最新版的数据，共有NA编号开头的1182个人，HG开头的1768个人！
<a href="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/phase3/data/">ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/phase3/data/</a>
也可以按照人种来查看这些数据：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data/</a>
每个人的目录下面都有 四个数据文件夹
Oct 01 2014 00:00    Directory alignment
Oct 01 2014 00:00    Directory exome<em>alignment
Oct 01 2014 00:00    Directory high</em>coverage<em>alignment
Oct 01 2014 00:00    Directory sequence</em>read
这些数据实在是太丰富了！
也可以直接看最新版的vcf文件，记录了这两千多人的所有变异位点信息！
可以直接看到所有的位点，具体到每个人在该位点是否变异！
<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/</a>
不过它的基因型信息是通过MVNcall+SHAPEIT这个程序call出来的，具体原理见：<a href="http://www.ncbi.nlm.nih.gov/pubmed/23093610">http://www.ncbi.nlm.nih.gov/pubmed/23093610</a>
而且网站还提供一些教程：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/working/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/working/</a>
我们肯定可以在千人基因计划的官网下载测序数据，主要是vcf格式的突变！
Coriell Catalog website: 1000 Genomes Project
1000 Genomes website: browser.1000genomes.org/index.html (by SNP ID)
1000 Genomes website: www.1000genomes.org/data (bulk data)
但是关于它的表达数据，就不是那么简单了！
The most important available existing expression datasets involving 1000g individuals are probably the following:</p>

<p>RNAseq (mRNA &amp; miRNA) on 465 individuals (CEU, TSI, GBR, FIN, YRI)</p>

<p>Pre-publication RNA-sequencing data from the Geuvadis project is available through <a href="http://www.geuvadis.org">http://www.geuvadis.org</a></p>

<p><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEUV-1/samples.html">http://www.ebi.ac.uk/arrayexpress/experiments/E-GEUV-1/samples.html</a>
<a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEUV-2/samples.html">http://www.ebi.ac.uk/arrayexpress/experiments/E-GEUV-2/samples.html</a></p>

<p>RNAseq on 60 CEU individual [1]</p>

<p><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-197">http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-197</a></p>

<p>Expression arrays on about 800 HapMap 3 individuals with a lot of overlap with 1000g data [1,2]</p>

<p><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-198">http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-198</a>
<a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-264">http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-264</a></p>

<p>RNAseq for 69 YRI individuals [3]</p>

<p><a href="http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-19480">http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-19480</a></p>

<hr />

<h1 id="居然可以下载千人基因组计划的所有数据bam-vcf数据">居然可以下载千人基因组计划的所有数据bam，vcf数据</h1>

<p>它有两个ftp站点存储所有的数据！
<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/</a>
<a href="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/">ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/</a>
直接看最新版的数据，共有NA编号开头的1182个人，HG开头的1768个人！
<a href="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/phase3/data/">ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/phase3/data/</a>
也可以按照人种来查看这些数据：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data/</a>
每个人的目录下面都有 四个数据文件夹</p>

<pre><code>Oct 01 2014 00:00    Directory alignment
Oct 01 2014 00:00    Directory exome_alignment
Oct 01 2014 00:00    Directory high_coverage_alignment
Oct 01 2014 00:00    Directory sequence_read
</code></pre>

<p>这些数据实在是太丰富了！
也可以直接看最新版的vcf文件，记录了这两千多人的所有变异位点信息！
可以直接看到所有的位点，具体到每个人在该位点是否变异！
<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/</a>
不过它的基因型信息是通过MVNcall+SHAPEIT这个程序call出来的，具体原理见：<a href="http://www.ncbi.nlm.nih.gov/pubmed/23093610">http://www.ncbi.nlm.nih.gov/pubmed/23093610</a>
而且网站还提供一些教程：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/working/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/working/</a></p>

<p>还有Illumina的450K甲基化芯片数据：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/hgsv_sv_discovery/working/20151214_450k_methylation/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/hgsv_sv_discovery/working/20151214_450k_methylation/</a>
还有一个小程序，<a href="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/technical/browser/vcf_to_ped_converter/version_1.1/">ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/technical/browser/vcf_to_ped_converter/version_1.1/</a>
还有Illumina的450K甲基化芯片数据：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/hgsv_sv_discovery/working/20151214_450k_methylation/">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/hgsv_sv_discovery/working/20151214_450k_methylation/</a>
还有一个小程序，<a href="ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/technical/browser/vcf_to_ped_converter/version_1.1/">ftp://ftp.ncbi.nlm.nih.gov/1000genomes/ftp/technical/browser/vcf_to_ped_converter/version_1.1/</a></p>

<hr />

<h1 id="roadmap">roadmap</h1>

<p><a href="http://www.roadmapepigenomics.org/">http://www.roadmapepigenomics.org/</a></p>

<p>精选的129个细胞系，细胞系的介绍如下：<a href="http://www.broadinstitute.org/~anshul/projects/roadmap/metadata/EID_metadata.tab">http://www.broadinstitute.org/~anshul/projects/roadmap/metadata/EID_metadata.tab</a></p>

<p>对每个细胞系，都至少处理了5个核心组蛋白修饰数据，还有其它若干转录因子数据。
官网介绍的很详细，我就不翻译了：</p>

<p>The NIH Roadmap Epigenomics Mapping Consortium was launched with the goal of producing a public resource of human epigenomic data to catalyze basic biology and disease-oriented research. The Consortium leverages experimental pipelines built around next-generation sequencing technologies to map DNA methylation, histone modifications, chromatin accessibility and small RNA transcripts in stem cells and primary ex vivo tissues selected to represent the normal counterparts of tissues and organ systems frequently involved in human disease. The Consortium expects to deliver a collection of normal epigenomes that will provide a framework or reference for comparison and integration within a broad array of future studies. The Consortium also aims to close the gap between data generation and its public dissemination by rapid release of raw sequence data, profiles of epigenomics features and higher-level integrated maps to the scientific community. The Consortium is also committed to the development, standardization and dissemination of protocols, reagents and analytical tools to enable the research community to utilize, integrate and expand upon this body of data.</p>

<p>首先是这个网站：
<a href="http://www.encode-roadmap.org/">http://www.encode-roadmap.org/</a>
矩阵很容易看懂roadmap处理了哪些细胞系，进行了什么样的处理，数据可以直接下载。</p>

<p>然后我比较首先推崇broad研究所的下载方式</p>

<p>里面还列出了他们用过的peaks caller 工具：
<a href="http://www.broadinstitute.org/~anshul/projects/encode/preprocessing/peakcalling/">http://www.broadinstitute.org/~anshul/projects/encode/preprocessing/peakcalling/</a>  可以看到，主要有MACS，peakranger，quest，sicer，peakseq，hotspot等等
直接进入broad分析好的peaks结果：</p>

<pre><code>[DIR]   Parent Directory    
-   
[DIR]   broadPeak/  08-Feb-2015 21:00   -   
[DIR]   gappedPeak/ 08-Feb-2015 21:00   -   
[DIR]   lowq/   31-Aug-2014 20:42   -   
[DIR]   narrowPeak/ 08-Feb-2015 20:59   -   
</code></pre>

<p>这里面有3种peaks，我现在还没有搞懂是什么意思。</p>

<p>接着是 iHEC存放的数据：</p>

<p><a href="http://epigenomesportal.ca/ihec/download.html">http://epigenomesportal.ca/ihec/download.html</a>
我还是第一次看到这个数据接口，也是以文件夹文件的形式直接浏览，根据自己的需求下载即可：
除了ENCODE计划的数据，还有Blueprint计划和roadmap计划的数据都可以下载。
NIH Roadmap 2014-05-29  Click here for policies
最后可以从圣路易斯华盛顿大学里面下载</p>

<p>圣路易斯华盛顿大学Washington University in St. Louis，简称（Wash U，WU）以美国国父乔治·华盛顿命名，始建于1853年2月22日，位于美国密苏里州圣路易斯市，是美国历史上建校最早也是最负盛名的“华盛顿大学”，该校在美国新闻和世界报道（US News &amp; World Report）2014大学综合排名中名列14位。
里面有一个非常详细的页面来介绍roadmap的各种数据:<a href="http://egg2.wustl.edu/roadmap/web_portal/processed_data.html">http://egg2.wustl.edu/roadmap/web_portal/processed_data.html</a>
如果你已经了解了roadmap计划，就很容易找到自己的数据，从而直接浏览器或者wget下载即可。
首先是序列比对结果下载。
onsolidated Epigenomes:36 bp mappability filtered, pooled and subsampled read alignment files:
<a href="http://egg2.wustl.edu/roadmap/data/byFileType/alignments/consolidated/">http://egg2.wustl.edu/roadmap/data/byFileType/alignments/consolidated/</a>
Unconsolidated Epigenomes (Uniform mappability): 36 bp mappability filtered primary alignment files:
<a href="http://egg2.wustl.edu/roadmap/data/byFileType/alignments/unconsolidated/">http://egg2.wustl.edu/roadmap/data/byFileType/alignments/unconsolidated/</a>
包括各种peaks记录文件下载
Narrow contiguous regions of enrichment (peaks) for histone ChIP-seq and DNase-seq
Data format: NarrowPeak
<a href="http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated/narrowPeak/">http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated/narrowPeak/</a>
Broad domains on enrichment for histone ChIP-seq and DNase-seq)
Data format: BroadPeak
<a href="http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated/broadPeak/">http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated/broadPeak/</a></p>

<p>Data format: GappedPeak (subset of domains containing at least one narrow peaks)
<a href="http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated/gappedPeak/">http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated/gappedPeak/</a></p>


  <footer>
  
<nav class="post-nav">
  <span class="nav-prev">&larr; <a href="/blog/cn/2017/11/nat123/">nat123</a></span>
  <span class="nav-next"><a href="/blog/cn/2017/11/data_format/">常用生物信息学格式介绍</a> &rarr;</span>
</nav>





<script src="//yihui.name/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>

<script async src="//yihui.name/js/center-img.js"></script>

  



<script src="//cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.12.0/languages/tex.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



  
  <hr>
  <div class="copyright">&copy; <a href="xxxx">Who care</a> 2017 - Forever, maybe</div>
  
  </footer>
  </article>
  
  </body>
</html>

